clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java

```diff
@@ -83,16 +83,89 @@ public PartitionData(long offset, long timestamp, String metadata) {
         public PartitionData(long offset, String metadata) {
             this(offset, DEFAULT_TIMESTAMP, metadata);
         }
+
+        @Override
+        public String toString() {
+            StringBuilder bld = new StringBuilder();
+            bld.append("{timestamp: ").append(timestamp).
+                append(", offset: ").append(offset).
+                append(", metadata: ").append(metadata).
+                append("}");
+            return bld.toString();
+        }
+    }
+
+    public static class Builder extends AbstractRequest.Builder<OffsetCommitRequest> {
+        private String groupId;
+        private String memberId = DEFAULT_MEMBER_ID;
+        private int generationId = DEFAULT_GENERATION_ID;
+        private long retentionTime = DEFAULT_RETENTION_TIME;
+        private Map<TopicPartition, PartitionData> offsetData;
+
+        public Builder(String groupId, Map<TopicPartition, PartitionData> offsetData) {
+            super(ApiKeys.OFFSET_COMMIT);
+            this.groupId = groupId;
+            this.offsetData = offsetData;
+        }
+
+        Builder setGroupId(String groupId) {
+            this.groupId = groupId;
+            return this;
+        }
+
+        public Builder setMemberId(String memberId) {
+            this.memberId = memberId;
+            return this;
+        }
+
+        public Builder setGenerationId(int generationId) {
+            this.generationId = generationId;
+            return this;
+        }
+
+        public Builder setRetentionTime(long retentionTime) {
+            this.retentionTime = retentionTime;
+            return this;
+        }
+
+        @Override
+        public OffsetCommitRequest build() {
+            short version = version();
+            switch (version) {
+                case 0:
+                    return new OffsetCommitRequest(groupId, offsetData);
+                case 1:
+                    return new OffsetCommitRequest(groupId, generationId, memberId,
+                            offsetData);
+                case 2:
+                    return new OffsetCommitRequest(groupId, generationId, memberId,
+                            retentionTime, offsetData, version);
+                default:
```

Comment:
Hmm.  Just to clarify, you think that throwing the exception when we see an unexpected version is the right way to go?  I can add that to the other requests if so.